src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -0,0 +1,474 @@
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "0 2");
+        }
+
+        [Fact]
+        public void Return()
+        {
+            string source = @"
+class C
+{
+    static int M()
+    {
+        return default;
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void BadReturn()
+        {
+            string source = @"
+class C
+{
+    static int? M()
+    {
+        return default;
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions);
+            comp.VerifyDiagnostics(
+                // (6,16): error CS8300: Cannot convert default to 'int?' because it is a value or nullable type
+                //         return default;
+                Diagnostic(ErrorCode.ERR_RefCantBeDefault, "default").WithArguments("int?").WithLocation(6, 16)
+                );
+        }
+
+        [Fact]
+        public void ConstAndProperty()
+        {
+            string source = @"
+class C
+{
+    const int x = default;
+    static int P { get { return default; } }
+    static void Main()
+    {
+        System.Console.Write($""{x} {P}"");
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "0 0");
+        }
+
+        [Fact]
+        public void Generic()
+        {
+            string source = @"
+class C1<T>
+{
+    static void M()
+    {
+        T t = default;
+    }
+}
+class C2<T> where T : struct
+{
+    static void M()
+    {
+        T t = default;
+    }
+}
+";
+            // PROTOTYPE(default) The constrained T should be ok to infer "default" from
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions);
+            comp.VerifyDiagnostics(
+                // (13,14): error CS8300: Cannot convert default to 'T' because it is a value or nullable type
+                //        T t = default;
+                Diagnostic(ErrorCode.ERR_RefCantBeDefault, "default").WithArguments("T").WithLocation(13, 14),
+                // (6,14): error CS8300: Cannot convert default to 'T' because it is a value or nullable type
+                //        T t = default;
+                Diagnostic(ErrorCode.ERR_RefCantBeDefault, "default").WithArguments("T").WithLocation(6, 14)
+                );
+        }
+
+        [Fact]
+        public void DynamicInvocation()
+        {
+            string source = @"
+class C
+{
+    static void M1()
+    {
+        dynamic d = null;
+        d.M2(default);
+    }
```

Comment:
I think we may need to forbid this (until and unless we build runtime support for it). It is like passing a method group.